core/ts/MaciState.ts

```diff
@@ -717,6 +728,169 @@ class Poll {
         return this.numBatchesTallied * batchSize < this.ballots.length
     }
 
+    public hasUnfinishedSubsidyCalculation = () => {
+        const batchSize = this.batchSizes.subsidyBatchSize
+        return (this.rbi * batchSize < this.ballots.length) && (this.cbi * batchSize < this.ballots.length)
+    }
+
+    public subsidyPerBatch = (): BigInt[] => {
+        const batchSize = this.batchSizes.subsidyBatchSize
+
+        assert(this.hasUnfinishedSubsidyCalculation(), 'No more subsidy batches to calculate')
+
+        const stateRoot = this.stateTree.root
+        const ballotRoot = this.ballotTree.root
+        const sbSalt = this.sbSalts[this.currentMessageBatchIndex]
+        const sbCommitment = hash3([stateRoot, ballotRoot, sbSalt])
+
+        const currentSubsidy = this.subsidy.map((x) => BigInt(x.toString()))
+        let currentSubsidyCommitment  = BigInt(0)
+        let currentSubsidySalt = BigInt(0)
+        let saltIndex = this.previousSubsidyIndexToString()
+        console.log(`prevIdx=${saltIndex}, curIdx=${this.rbi}-${this.cbi}`)
+        if (this.rbi !== 0 || this.cbi !== 0) {
+            currentSubsidySalt = BigInt(this.subsidySalts[saltIndex])
+            currentSubsidyCommitment = BigInt(genTallyResultCommitment(this.subsidy, currentSubsidySalt, this.treeDepths.voteOptionTreeDepth))
```

Comment:
```suggestion
            currentSubsidyCommitment = BigInt(genTallyResultCommitment(this.subsidy, currentSubsidySalt, this.treeDepths.voteOptionTreeDepth).valueOf())
```
[The valueOf() method returns the wrapped primitive value of a BigInt object.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/valueOf)